kubert/src/runtime.rs

```diff
@@ -126,6 +205,12 @@ impl<S> Builder<S> {
         self
     }
 
+    /// Configures the runtime to record watch metrics with the given registry
+    pub fn with_metrics(mut self, registry: &mut Registry) -> Self {
+        self.metrics = Some(ResourceWatchMetrics::new(registry));
+        self
+    }
```

Comment:
My recent experience in the proxy tells me that we probably want to decouple metrics registration and builder initialization (so that we can build the admin server before calling this).

To accomplish this I recommend that we expose a single public `kubert::RuntimeMetrics` type with a `pub fn register`.

```rust
pub struct RuntimeMetrics {
    watch: ResourceWatchMetrics
}
```

```suggestion
    /// Configures the runtime to record watch metrics with the given registry
    pub fn with_metrics(mut self, m: RuntimeMetrics) -> Self {
        self.metrics = Some(m);
        self
    }
```